tests/v2/codecoverage/setuprancher/main.go

```diff
@@ -0,0 +1,505 @@
+	if err != nil {
+		return err
+	}
+
+	updatedDeployment.Spec.Template.Spec.Containers[0].Args = []string{}
+	updatedDeployment.Spec.Template.Spec.Containers[0].Image = rancherTestImage
+	updatedDeployment.Spec.Strategy.RollingUpdate = nil
+	updatedDeployment.Spec.Strategy.Type = appv1.RecreateDeploymentStrategyType
+
+	unstructuredResp, err := deploymentResource.Update(context.TODO(), unstructured.MustToUnstructured(updatedDeployment), metav1.UpdateOptions{})
+	if err != nil {
+		return err
+	}
+
+	err = kwait.Poll(500*time.Millisecond, 2*time.Minute, func() (done bool, err error) {
+		_, err = deploymentResource.Get(context.TODO(), deploymentName, metav1.GetOptions{})
+		if err != nil {
+			return false, nil
+		}
+		return true, nil
+	})
+	if err != nil {
+		return err
+	}
+
+	newDeployment := &appv1.Deployment{}
+	err = scheme.Scheme.Convert(unstructuredResp, newDeployment, unstructuredResp.GroupVersionKind())
+	if err != nil {
+		return err
+	}
+
+	// wait until pods are ready with new images
+
+	result, err := deploymentResource.Watch(context.TODO(), metav1.ListOptions{
+		FieldSelector:  "metadata.name=" + deploymentName,
+		TimeoutSeconds: &defaults.WatchTimeoutSeconds,
+	})
+	if err != nil {
+		return err
+	}
+
+	err = wait.WatchWait(result, func(event watch.Event) (ready bool, err error) {
+		deploymentUnstructured := event.Object.(*k8sUnstructured.Unstructured)
+		newDeployment := &appv1.Deployment{}
+		err = scheme.Scheme.Convert(deploymentUnstructured, newDeployment, deploymentUnstructured.GroupVersionKind())
+		if err != nil {
+			return false, err
+		}
+		if newDeployment.Status.ReadyReplicas == int32(3) {
+			return true, nil
+		}
+		return false, nil
+	})
+
+	if err != nil {
+		return err
+	}
+
+	err = kwait.Poll(500*time.Millisecond, 2*time.Minute, func() (done bool, err error) {
+		_, err = deploymentResource.Get(context.TODO(), deploymentName, metav1.GetOptions{})
+		if k8sErrors.IsInternalError(err) || k8sErrors.IsServiceUnavailable(err) {
+			return false, nil
+		}
+		return true, nil
+	})
+	if err != nil {
+		return err
+	}
+
+	err = kwait.Poll(500*time.Millisecond, 10*time.Minute, func() (done bool, err error) {
+		webhookDeployment, err := deploymentResource.Get(context.TODO(), "rancher-webhook", metav1.GetOptions{})
+		if err != nil {
+			return false, nil
+		}
+		newDeployment := &appv1.Deployment{}
+		err = scheme.Scheme.Convert(webhookDeployment, newDeployment, webhookDeployment.GroupVersionKind())
+		if err != nil {
+			return false, err
+		}
+		if newDeployment.Status.ReadyReplicas == int32(1) {
+			return true, nil
+		}
+		return false, nil
+	})
+
+	if err != nil {
+		return err
+	}
+
+	deploymentResource = dynamicClient.Resource(deployments.DeploymentGroupVersionResource).Namespace("cattle-fleet-local-system")
+	err = kwait.Poll(500*time.Millisecond, 10*time.Minute, func() (done bool, err error) {
+		webhookDeployment, err := deploymentResource.Get(context.TODO(), "fleet-agent", metav1.GetOptions{})
```

Comment:
Really small nit, maybe changing these names for the deployments watch might increase readability a bit, maybe calling `deployment` inside the polls? Applicable for below polls as well